src/linalg/muladd.jl

```diff
@@ -47,54 +35,83 @@ length(M::MulAdd) = prod(size(M))
 size(M::MulAdd) = length.(axes(M))
 axes(M::MulAdd) = axes(M.C)
 
+similar(M::MulAdd, ::Type{T}, axes) where {T,N} = Array{T}(undef, length.(axes))
+similar(M::MulAdd, ::Type{T}) where T = similar(M, T, axes(M))
+similar(M::MulAdd) = similar(M, eltype(M))
+
+function instantiate(M::MulAdd)
+    @boundscheck check_mul_axes(M.α, M.A, M.B)
+    @boundscheck check_mul_axes(M.β, M.C)
+    @boundscheck axes(M.A,1) == axes(M.C,1) || throw(DimensionMismatch("First axis of A, $(axes(A,1)), and first axis of C, $(axes(C,1)) must match"))
+    @boundscheck axes(M.B,2) == axes(M.C,2) || throw(DimensionMismatch("Second axis of B, $(axes(B,2)), and second axis of C, $(axes(C,2)) must match"))
+    M
+end
+
+struct Lmul{StyleA, StyleB, TypeA, TypeB}
+    A::TypeA
+    B::TypeB
+end
+
+Lmul(A::TypeA, B::TypeB) where {TypeA,TypeB} = Lmul{typeof(MemoryLayout(TypeA)),typeof(MemoryLayout(TypeB)),TypeA,TypeB}(A,B)
+
+
 BroadcastStyle(::Type{<:MulAdd}) = ApplyBroadcastStyle()
 BroadcastStyle(::Type{<:Lmul}) = ApplyBroadcastStyle()
 
 struct LmulStyle <: AbstractArrayApplyStyle end
-abstract type AbstractMulAddStyle <: AbstractArrayApplyStyle end
-struct MulAddStyle <: AbstractMulAddStyle end
-
+struct MulAddStyle <: AbstractArrayApplyStyle end
+struct ScalarMulStyle <: ApplyStyle end
+struct IdentityMulStyle <: AbstractArrayApplyStyle end
+
+combine_mul_styles() = MulAddStyle()
+combine_mul_styles(_) = IdentityMulStyle()
+combine_mul_styles(::ScalarLayout) = ScalarMulStyle()
+combine_mul_styles(c1, c2) = result_mul_style(combine_mul_styles(c1), combine_mul_styles(c2))
+@inline combine_mul_styles(c1, c2, cs...) = result_mul_style(combine_mul_styles(c1), combine_mul_styles(c2, cs...))
+
+# result_mul_style works on types (singletons and pairs), and leverages `Style`
+result_mul_style(s::ApplyStyle) = s
```

Comment:
Is single-argument `result_mul_style` used?

Reply:
No. Again, this is copy-and-pasted, but I'll delete it to avoid confusion.